### PR TITLE
Add `init` accessors in `readonly struct`s and `readonly` properties

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/csharplang/issues/3884.

### CODE REVIEW

File: proposals/csharp-9.0/init.md [hunk 1]
```diff
@@ -242,6 +242,27 @@ Restrictions of this feature:
 - All overrides of a property must have `init` if the base had `init`. This rule
 also applies to interface implementation.
 
+### Readonly structs
+
+`init` accessors (both auto-implemented accessors and manually-implemented
```
Comment (pkulikov) [inline proposals/csharp-9.0/init.md:247]: As for manually-implemented accessors, please see https://github.com/dotnet/csharplang/issues/3884#issuecomment-714987528

File: proposals/csharp-9.0/init.md [hunk 2]
```diff
@@ -242,6 +242,27 @@ Restrictions of this feature:
...
+accessors) are permitted on properties of `readonly struct`s, as well as
+`readonly` properties. `init` accessors are not permitted to be marked
+`readonly` themselves.
```
Comment (jaredpar) [inline proposals/csharp-9.0/init.md:250]: Is the last statement true for all `struct` types or only non-readonly `struct`? Think we should clarify that point.

**Review by jaredpar — COMMENTED** [review]
Minor feedback

**Comment by 333fred** [comment]
@jaredpar addressed feedback.

**Review by jaredpar — APPROVED** [review]

**Pull request merged**